src/main/java/seedu/address/ui/PersonCard.java

```diff
@@ -104,58 +93,46 @@ public PersonCard(Person person, int displayedIndex) {
-                priorityShape.setBackground(new Background(new BackgroundFill(ColorPicker.ORANGE_DARK,
-                        CornerRadii.EMPTY, Insets.EMPTY)));
+                priorityShape.setBorder(PriorityIndicatorComponent.ARCHIVED_MEDIUM_PRIORITY_BORDER);
+                priorityShape.setBackground(PriorityIndicatorComponent.ARCHIVED_MEDIUM_PRIORITY_BACKGROUND);
             }
             break;
         case "high":
             if (!person.getIsArchive()) {
-                priorityShape.setBorder(new Border(new BorderStroke(ColorPicker.RED_BORDER,
-                        BorderStrokeStyle.SOLID, CornerRadii.EMPTY, new BorderWidths(BORDER_SIZE))));
-                priorityShape.setBackground(new Background(new BackgroundFill(ColorPicker.RED, CornerRadii.EMPTY,
-                        Insets.EMPTY)));
+                priorityShape.setBorder(PriorityIndicatorComponent.UNARCHIVED_HIGH_PRIORITY_BORDER);
+                priorityShape.setBackground(PriorityIndicatorComponent.UNARCHIVED_HIGH_PRIORITY_BACKGROUND);
             } else {
-                priorityShape.setBorder(new Border(new BorderStroke(ColorPicker.RED_BORDER_DARK,
-                        BorderStrokeStyle.SOLID, CornerRadii.EMPTY, new BorderWidths(BORDER_SIZE))));
-                priorityShape.setBackground(new Background(new BackgroundFill(ColorPicker.RED_DARK,
-                        CornerRadii.EMPTY, Insets.EMPTY)));
+                priorityShape.setBorder(PriorityIndicatorComponent.ARCHIVED_HIGH_PRIORITY_BORDER);
+                priorityShape.setBackground(PriorityIndicatorComponent.ARCHIVED_HIGH_PRIORITY_BACKGROUND);
             }
             break;
         case "undefined":
             if (!person.getIsArchive()) {
-                priorityShape.setBorder(new Border(new BorderStroke(ColorPicker.WHITE_BORDER,
-                        BorderStrokeStyle.SOLID, CornerRadii.EMPTY, new BorderWidths(BORDER_SIZE))));
-                priorityShape.setBackground(new Background(new BackgroundFill(Color.WHITE, CornerRadii.EMPTY,
-                        Insets.EMPTY)));
+                priorityShape.setBorder(PriorityIndicatorComponent.UNARCHIVED_UNDEFINED_PRIORITY_BORDER);
+                priorityShape.setBackground(PriorityIndicatorComponent.UNARCHIVED_UNDEFINED_PRIORITY_BACKGROUND);
             } else {
-                priorityShape.setBorder(new Border(new BorderStroke(ColorPicker.WHITE_BORDER_DARK,
-                        BorderStrokeStyle.SOLID, CornerRadii.EMPTY, new BorderWidths(BORDER_SIZE))));
-                priorityShape.setBackground(new Background(new BackgroundFill(ColorPicker.WHITE_DARK,
-                        CornerRadii.EMPTY, Insets.EMPTY)));
+                priorityShape.setBorder(PriorityIndicatorComponent.ARCHIVED_UNDEFINED_PRIORITY_BORDER);
+                priorityShape.setBackground(PriorityIndicatorComponent.ARCHIVED_UNDEFINED_PRIORITY_BACKGROUND);
```

Comment:
I think this improvement is good because there is better abstraction, so PersonCard doesn't need to know what color is being assigned.
I was thinking if creating a new PriorityShape object might be better (more abstract)
e.g. `priorityShape = new PriorityShape(priority = "high", isArchive = true);`
PriorityShape has 2 fields: PriorityBorder and PriorityBackground; both are classes with take in priority = "high", isArchive = true 
In each one of them, e.g. PriorityBorder, there is a `Color` field, and a switch case (8 cases) that chooses the color ONLY. Then finally we return a PriorityBorder with the required fields. This might resolve the code duplication in the PriorityIndicatorComponent class.
```
ColorPicker color; // field
....
/** constructor **/
switch:
case "high":
  color = !isArchive ? ColorPicker.RED_BORDER ? ColorPicker.RED_BORDER_DARK;
case ...

Border getBorder():
  return new Border(new BorderStroke(
            color,
            BorderStrokeStyle.SOLID,
            CornerRadii.EMPTY,
            new BorderWidths(BORDER_SIZE)));
```
P.S.
You may even may this more abstract by having a generic Shape and Background class that takes in color = "red" and isDark = true, and PriorityShape is the one that does the conversion of priority to color.
You could also have the Color class to take in the color and isDark fields as well, and allow it to be the one handling the different cases, instead of doing it in the Border

Reply:
The thing is this PS object would have to be a child class of the Hbox class. Then if you realise I am not really creating a new priorityshape object due to the fxml file being used to create it. 

If we use the Pborder and Pbackground class, I was concerned that there would be duplication of code. (like the repeat of the switch cases). But yes, I quite agree with this idea. though im wondering if it would be relevant if we do not create a new priorityshape object